### PR TITLE
Add Domain Block (IOS-5)

### LINKED ISSUE #364: Missing block domain dropdown option on profiles

### PR DESCRIPTION
**Pull request by zeitschlag**
- Add a new Menu-entry on Profile-Screen only to block/unblock entire domains
- At the moment it's for `MastodonUser` only as #1192 still needs some time, love and labour (but this PR will affect me and my work at #1192, at least slightly)
- Lots of stuff was already prepared.
- Did someone say screenshots? Screenshots (the last one is there to show that all `botsin.space` are blocked):
blockdomain_screenshots

Will fix #364

### CODE REVIEW

File: Mastodon/Scene/Profile/ProfileViewController.swift [hunk 1]
```diff
@@ -829,6 +830,27 @@ extension ProfileViewController: ProfileHeaderViewControllerDelegate {
                 let cancelAction = UIAlertAction(title: L10n.Common.Controls.Actions.cancel, style: .cancel, handler: nil)
                 alertController.addAction(cancelAction)
                 present(alertController, animated: true, completion: nil)
+            case .domainBlocking:
+                    guard let user = viewModel.user else { return }
+                    let domain = user.domainFromAcct
+
+                    let alertController = UIAlertController(
+                        title: "Unblock domain \(domain)",
+                        message: "Really unblock domain \(domain)?!",
```
Comment (DK1DA) [inline Mastodon/Scene/Profile/ProfileViewController.swift]: Here `L10n` isn't used yet.

File: MastodonSDK/Sources/MastodonUI/ViewModel/RelationshipViewModel.swift [hunk 2]
```diff
@@ -263,6 +268,10 @@ extension RelationshipViewModel {
             optionSet.insert(.showReblogs)
         }
 
+        if isDomainBlocking {
+            optionSet.insert(.domainBlocking)
```
Comment (DK1DA) [inline MastodonSDK/Sources/MastodonUI/ViewModel/RelationshipViewModel.swift:272]: I'm wondering if we, somewhere need to remove this attribute incase a domain gets unblocked? However the App likely won'T notice this anyway properly so maybe this is going away with the `MastodonUser` refactoring anyways.

**Review by kimar — APPROVED** [review]
LGTM, just one remark regarding missing `L10n`.

File: Mastodon/Scene/Profile/ProfileViewController.swift (hunk 1 above)
Comment (zeitschlag) [reply Mastodon/Scene/Profile/ProfileViewController.swift]: Oh, good catch, thank you! 🤦

**Pull request merged**